File: src/test/java/org/blitzortung/android/map/overlay/color/StrikeColorHandlerTest.java
package org.blitzortung.android.map.overlay.color;

import android.content.SharedPreferences;
import org.blitzortung.android.app.view.PreferenceKey;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.robolectric.RobolectricTestRunner;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.*;

@RunWith(RobolectricTestRunner.class)
public class StrikeColorHandlerTest {
    @Mock
    private SharedPreferences sharedPreferences;

    private StrikeColorHandler strikeColorHandler;

    @Before
    public void setUp()
    {
        MockitoAnnotations.initMocks(this);

        when(sharedPreferences.getString(PreferenceKey.MAP_TYPE.toString(), ColorTarget.SATELLITE.toString())).thenReturn(ColorTarget.SATELLITE.toString());
        when(sharedPreferences.getString(PreferenceKey.COLOR_SCHEME.toString(), ColorScheme.BLITZORTUNG.toString())).thenReturn(ColorScheme.BLITZORTUNG.toString());

        strikeColorHandler = spy(new StrikeColorHandler(sharedPreferences));
    }

    @Test
    public void testSatelliteColors()
    {
        int[] colors = strikeColorHandler.getColors(ColorTarget.SATELLITE);

        assertThat(colors.length, is(6));
        assertThat(colors, is(new int[]{0xffe4f9f9, 0xffd8f360, 0xffdfbc51, 0xffe48044, 0xffe73c3b, 0xffb82e2d}));
    }

    @Test
    public void testMapColors()
    {
        int[] colors = strikeColorHandler.getColors(ColorTarget.STREETMAP);

        assertThat(colors.length, is(6));
        assertThat(colors, is(new int[]{0, 0, 0, 0, 0, 0}));

        verify(strikeColorHandler, times(1)).modifyBrightness(eq(new int[]{0xffe4f9f9, 0xffd8f360, 0xffdfbc51, 0xffe48044, 0xffe73c3b, 0xffb82e2d}), anyFloat());
    }
}

File: app/src/test/java/org/blitzortung/android/map/overlay/color/StrikeColorHandlerTest.kt
package org.blitzortung.android.map.overlay.color

import android.content.Context
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(manifest = Config.NONE)
class StrikeColorHandlerTest {

    lateinit var strikeColorHandler: StrikeColorHandler

    @Before
    fun setUp() {
        val context = RuntimeEnvironment.getApplication()
        val preferences = context.getSharedPreferences(context.packageName, Context.MODE_PRIVATE)

        strikeColorHandler = StrikeColorHandler(preferences)

    }

    @Test
    fun shouldReturnProperSatelliteColors() {
        val colors = strikeColorHandler.getColors(ColorTarget.SATELLITE)

        assertThat(colors).isEqualTo(ColorScheme.BLITZORTUNG.strikeColors)
    }

    @Test
    fun shouldReturnProperSatelliteTextColor() {
        val textColor = strikeColorHandler.getTextColor(ColorTarget.SATELLITE)

        assertThat(textColor).isEqualTo(0xff000000.toInt())
    }

    @Test
    fun shouldReturnProperStreetmapColors() {
        val colors = strikeColorHandler.getColors(ColorTarget.STREETMAP)

        assertThat(colors).containsExactly(
            0xffcccccc.toInt(),
            0xffcccc8d.toInt(),
            0xffccb361.toInt(),
            0xffcc8d4d.toInt(),
            0xffc0664d.toInt(),
            0xffb34040.toInt(),
        )
    }

    @Test
    fun shouldReturnProperStreetmapTextColor() {
        val textColor = strikeColorHandler.getTextColor(ColorTarget.STREETMAP)

        assertThat(textColor).isEqualTo(0xffffffff.toInt())
    }
}